lib/name_drop/client.rb

```diff
@@ -66,23 +66,27 @@ def post(endpoint, attributes)
     # @param [Hash] attributes
     # @return [Hash] encapsulates Mention API response of PUT request
     def put(endpoint, attributes)
-      request = Net::HTTP::Put.new request_uri(endpoint)
-      request.body = JSON.dump(attributes)
-      execute_request request
+      execute_request build_request(endpoint, 'Put', attributes)
     end
 
     # Makes DELETE Request through RestClient::Request.execute
     #
     # @param [String] endpoint
     # @return [Hash] encapsulates Mention API response of DELETE request
     def delete(endpoint)
-      execute_request Net::HTTP::Delete.new(request_uri(endpoint))
+      execute_request build_request(endpoint, 'Delete')
     end
 
     # @!endgroup
 
     private
 
+    def build_request(endpoint, method, attributes = {})
+      request = "Net::HTTP::#{method}".constantize.new request_uri(endpoint)
+      request.body = JSON.dump(attributes) if attributes.any?
```

Comment:
I thought you needed to set the query params for a GET request on the URI object, like in the [docs](http://ruby-doc.org/stdlib-2.3.1/libdoc/net/http/rdoc/Net/HTTP.html), but it looks like this also works, just that params aren't present in the URL.